Rakefile

```diff
@@ -42,11 +42,15 @@ task :debug do
     openssl_version_number_str = OpenSSL::OPENSSL_VERSION_NUMBER.to_s(16)
     libressl_version_number_str = (defined? OpenSSL::LIBRESSL_VERSION_NUMBER) ?
       OpenSSL::LIBRESSL_VERSION_NUMBER.to_s(16) : "undefined"
+    providers_str = (defined? OpenSSL::Provider) ?
+      OpenSSL::Provider.provider_names.join(", ") : "undefined"
     puts <<~MESSAGE
       OpenSSL::OPENSSL_VERSION: #{OpenSSL::OPENSSL_VERSION}
       OpenSSL::OPENSSL_LIBRARY_VERSION: #{OpenSSL::OPENSSL_LIBRARY_VERSION}
       OpenSSL::OPENSSL_VERSION_NUMBER: #{openssl_version_number_str}
       OpenSSL::LIBRESSL_VERSION_NUMBER: #{libressl_version_number_str}
+      FIPS enabled: #{OpenSSL.fips_mode}
```

Comment:
Please note that the terminology "FIPS mode" was used in OpenSSL 1.0.x with the C APIs `FIPS_mode()` and `FIPS_mode_set()`. However, the terminology "FIPS mode" is not used anymore in OpenSSL 3. So, I think we need to remove some parts of the "fips mode" used in the comment. You can check https://github.com/openssl/openssl/discussions/21797 for details. And the "FIPS enabled" comes from the `EVP_default_properties_is_fips_enabled` used below.

https://github.com/ruby/openssl/blob/bff06067cd3cf611b50cb76adc2ae40f9406cb1e/ext/openssl/ossl.c#L412